File: patterns/java/src/refactoring_guru/patterns/fasad/example/CodecFactory.java
package refactoring_guru.patterns.fasad.example;

public class CodecFactory {
    public static Codec extract(VideoFile file) {
        String type = file.getCodecType();
        if (type.equals("mp4")) {
            return new MPEG4CompressionCodec();
        }
        return new OggCompressionCodec();
    }
}

File: patterns/cs/src/RefactoringGuru/Facade/Example/SomeComplexMediaLibrary/CodecFactory.cs
using System;

namespace RefactoringGuru.Facade.Example.SomeComplexMediaLibrary
{
    class CodecFactory
    {
        public static ICodec Extract(VideoFile file)
        {
            String type = file.GetCodecType();
            if (type.Equals("mp4"))
            {
                Console.WriteLine("CodecFactory: extracting mpeg audio...");
                return new MPEG4CompressionCodec();
            }
            else
            {
                Console.WriteLine("CodecFactory: extracting ogg audio...");
                return new OggCompressionCodec();
            }
        }
    }
}
